tools/love-api.lua

```diff
@@ -146,6 +156,9 @@ local function buildFunction(func, node, typeName)
     for _, param in ipairs(func.variants[1].arguments or {}) do
         for paramName in param.name:gmatch '[%a_][%w_]*' do
             params[#params+1] = paramName
+			if param.type == 'table' then
+				param.default = isTableOptional(param.table)
```

Comment:
Don't modify the meta data

Reply:
I addressed your concern in the latest version. You can test the logic in the [online interpreter](https://www.lua.org/cgi-bin/demo) with the following code by changing the values in param variable:
```lua
local function isTableOptional(tbl)
	local optional = true
	for _, field in ipairs(tbl) do
		if field.default == nil then
			optional = nil
		end
	end
	return optional
end

local param = {
	type = 'table',
	default = nil,
	table = {
		{
			default = true,
		},{
			--default = false,
		}
	},
}

local a = (param.type == 'table' and isTableOptional(param.table) or param.default) == nil and '' or '?'

print(string.format("'%s'", a))
```

Reply:
I think you can just use an extra variable.